tests/ui/option_and_then_some.rs

```diff
@@ -9,10 +9,6 @@ pub fn main() {
     let _ = x.and_then(|o| Some(o + 1));
     // and an easy counter-example
     let _ = x.and_then(|o| if o < 32 { Some(o) } else { None });
-
-    // Different type
-    let x: Result<u32, &str> = Ok(1);
-    let _ = x.and_then(Ok);
```

Comment:
Could you give a reason to remove these tests (for example compile errors)?
These tests are as regression tests for option_and_then_some lint.

Reply:
The new result_and_then_ok lint changes this to

```
    // Different type
    let x: Result<u32, &str> = Ok(1);
    let _ = x;
```

I could add it back in, but how could I then disable the result_and_then_ok lint from changing this file?

Reply:
Add `#![deny(clippy::result_and_then_ok)]` to file I think.

Reply:
Will do

Reply:
It turned out I needed to add `#![allow(clippy::result_and_then_ok)]` to prevent Clippy from applying the lint.